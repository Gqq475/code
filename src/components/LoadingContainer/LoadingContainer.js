import React from 'react'
import classes from './LoadingContainer.scss'
import { Intent, ProgressBar } from '@blueprintjs/core'

type Props = {
  content: String
};
export class LoadingContainer extends React.Component {
  props: Props;

  render () {
    let content = this.props.content ? this.props.content : _.LOADING
    return (
      <div className={classes['LoadingContainer-container']}>
        <label className={classes['waitingIndicatorTxt']}>{content}</label>
        <ProgressBar className={classes['waitingIndicator']} intent={Intent.PRIMARY} />
      </div>
    )
  }
}

export default LoadingContainer
