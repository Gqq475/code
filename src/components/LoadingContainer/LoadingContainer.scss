.LoadingContainer-container {
  position: fixed;
  left: 0;
  top: 80px;
  bottom: 0;
  right: 0;
  background: #000;
  opacity: 0.8;
  //Set 30 because blueprint table use z-index 20 at most
  z-index: 30;

  .waitingIndicatorTxt {
    color: #FFFFFF;
    font-family: "Roboto", sans-serif;
    position: absolute;
    top: 45%;
    left: 45%;
    font-size: 20px;
  }

  .waitingIndicator {
    width: 25%;
    position: absolute;
    top: 50%;
    left: 35%;
    z-index: inherit;
  }
}
