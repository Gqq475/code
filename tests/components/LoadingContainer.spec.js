import React from 'react'
import LoadingContainer from 'components/LoadingContainer/LoadingContainer'

describe('(Component) LoadingContainer', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <LoadingContainer />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
