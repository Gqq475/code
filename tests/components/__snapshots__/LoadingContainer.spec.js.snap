// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) LoadingContainer render normally 1`] = `
<div
  className="LoadingContainer-container"
>
  <label
    className="waitingIndicatorTxt"
  />
  <Blueprint.ProgressBar
    className="waitingIndicator"
    intent={0}
  />
</div>
`;
